File: project_euler/function_library/discarded/prime_sieve.py
from numpy import *;
import time;
def prime_sieve(N):
    start=time.time();   
    a=array(range(1,N+1),dtype=uint64);
    index=2;
    while index<len(a) and a[index-1]**2<=N:
        tmp=a[index:];
        tmp=tmp[mod(tmp,a[index-1])!=0];
        a=concatenate([a[:index],tmp]);
        index+=1;
    t=time.time()-start;
    return a,t;        
        
        

File: project_euler/function_library/prime_sieve.m
function [a t]=prime_sieve(N)
t1=now;
a=1:N;
index=2;
while index<length(a) && a(index)^2<=N
	tmp=a(index+1:end);
	tmp=tmp(mod(tmp,a(index))~=0);
	a=[a(1:index) tmp];
	index=index+1;
end
t=run_time(t1);
end